swebench/harness/test_spec/test_spec.py

```diff
@@ -91,12 +94,18 @@ def env_image_key(self):
         hash_object.update(hash_key.encode("utf-8"))
         hash_value = hash_object.hexdigest()
         val = hash_value[:22]  # 22 characters is still very likely to be unique
+        if self.repo not in MAP_REPO_TO_EXT:
+            return f"sw.env.py.{self.arch}.{val}:{self.env_image_tag}"
         return f"sweb.env.{MAP_REPO_TO_EXT[self.repo]}.{self.arch}.{val}:{self.env_image_tag}"
 
     @property
     def instance_image_key(self):
-        key = f"sweb.eval.{self.arch}.{self.instance_id.lower()}:{self.instance_image_tag}"
-        if self.is_remote_image:
+        if self.repo not in MAP_REPO_TO_EXT:
```

Comment:
Hmm, this and line 97 seems to be specific to SWA/etc. Can't this be set from the input files?

Reply:
We updated the dataset to include `docker_image_root` and removed all specific SWA references. I also removed the other special cases, and tested that both SWE (with default arguments) and SWA (with `--namespace logicstarai` and  `--dataset_name LogicStar/SWA-Bench`) still work as expected both when building new docker images and using the DockerHub images.